File: src/compiler/typescript_to_rust/generators/ic_object/functions/msg_cycles_available.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionMsgCyclesAvailable(): Rust {
    return /* rust */ `
        fn _azle_ic_msg_cycles_available(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            let return_value: boa_engine::bigint::JsBigInt = ic_cdk::api::call::msg_cycles_available().into();
            Ok(return_value.into())
        }
    `;
}

File: src/compiler/rust/canister/src/ic/msg_cycles_available.rs
use wasmedge_quickjs::{Context, JsFn, JsValue};

pub struct NativeFunction;
impl JsFn for NativeFunction {
    fn call(context: &mut Context, this_val: JsValue, argv: &[JsValue]) -> JsValue {
        context
            .new_string(&ic_cdk::api::call::msg_cycles_available().to_string())
            .into()
    }
}
